src/AdsClient/AmsSocketConnection.cs

```diff
@@ -6,22 +6,24 @@
 
 namespace Viscon.Communication.Ads;
 
-internal class AmsSocketConnection
+internal sealed class AmsSocketConnection
 {
     private const int ReceiveTaskTimeout = 3000;
 
     private readonly Socket socket;
     private readonly IIncomingMessageHandler messageHandler;
     private readonly Task receiveTask;
+    private readonly SocketAwaitable socketAwaitable = new SocketAwaitable();
+    private readonly byte[] headerBuffer = new byte[AmsHeaderHelper.AmsTcpHeaderSize];
```

Comment:
As the `ReceiveLoop` handles one receive after another, so in a serialized manner, the awaitable and the buffer can be re-used.

Reply:
Yes, I only got to optimizing the sending side of things.